src/Service/RegenerateProductUrl.php

```diff
@@ -134,6 +134,19 @@ public function execute(?array $productIds = null, ?int $storeId = null, bool $v
                 if (count($newUrls)) {
                     $regeneratedForStore += $this->replaceUrls($newUrls, true);
                 }
+            } catch (\Magento\UrlRewrite\Model\Exception\UrlAlreadyExistsException $e) {
+                $this->log(sprintf(
+                    '<error>Couldn\'t insert duplicate URL rewrites for the following ' .
```

Comment:
/wtd update the log message to explain that even one error will cause the entire batch of ~500 URLs to be rolled back and fail